Scripts/UI/city_ui.gd

```diff
@@ -5,7 +5,16 @@ signal purchased_hero  # Hero list UI Update
 
 var city : City
 ## currently visiting hero, null if none
-var hero_army : ArmyForm
+## this is the hero who is trading with city, not the one who is inside
+## UPDATE -- TODO remove -- we do not need this -- better use WM.selected_hero
```

Comment:
"better use WM.selected_hero"
please don't,
using state from external source makes it completely unclear what the state of city interaction is and will lead to weird bugs
this needs a bit of an update but don't just use state from another object in this complex of an interaction

Reply:
also using WM directly makes it impossible to simulate city state
as singletons cannot get a copy

Reply:
selected hero is an ui not mechanical concept

hero_army is more mechanical as it says a given hero si currently in state of trading with this city
it might have cost them a movement point to start trading

Reply:
`city_ui.gd` is ui
`city.gd` is mechanical
`world_manager.gd` (or WM) is also something between ui and controller from MVC -- we could use additional pointer to citty in city_ui, but i think it introduces redundancy which we will need to control if it does not diverge from WM

I think we should decide if army doing a trade is a mechanical move or only UI convenience.